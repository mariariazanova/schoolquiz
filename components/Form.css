.quiz-form {
    width: 100%;
}
.quiz-details {
    display: flex;
}
.quiz-details .labels-list {
    display: flex;
    flex-direction: column;
    width: 150px;
}
.quiz-details .labels-list label {
    margin-bottom: 10px;
    
}

.quiz-details .inputs-list {
    display: flex;
    flex-direction: column;
    width: 500px;
}

.quiz-details .inputs-list input, .quiz-details .inputs-list select {
    margin-bottom: 10px;
    margin-top: 2px;
    
}