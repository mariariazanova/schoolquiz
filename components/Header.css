.quiz {
    width: 100%;
    padding-right: 15px;
    padding-left: 15px;
    margin-right: auto;
    margin-left: auto;
}
.quiz .header {
    display: flex;
    justify-content: space-between;
}

.quiz  .logo_img {
    width: 100px;


}

.quiz .registration_block {
    display:flex;
}

.quiz .registration_block div {
    padding: 10px;
    margin: 10px;
    background-color: orange;
    border-radius: 5px;
    color: #fff;

}
.quiz .registration_block div:last-child {
    margin-right: 0px;
}

.quiz .registration_block div:hover {
    cursor: pointer;
}    

li.page-item {
    flex: auto;
    text-align: center;

}


.page-item .page-link{
    background-color: #008966;
    border-color: #007c5d;
    position: relative;
    display: block;
    padding: 0.5rem 0.75rem;
    line-height: 1.25;
    color: #fff;
    border: 0 solid transparent;
    text-decoration: none;
}

.page-item.active .page-link {
    background-color: #00bc8c;
    border-color: #007c5d;
}

.quiz .main {
    width: 50%; 
    margin: 10px 10px 10px 0;
    padding: 10px;
    border-radius: 5px;
    background-color: lightgreen;
}

.new_form {
    width: 400px;
    height: 400px;
    background-color: orange;
    position: absolute;
    color: white;
    border-radius: 5px;
    text-align: center;
    vertical-align: center;
    top: calc(50% - 200px);
    left: 50%;
    margin-left: -200px;
    box-shadow: 500px -500px 0px 1600px rgba(0,0,0,0.5);
    z-index: 1000;


}

.new_form  .btn-submit {
    margin-top: 10px;
    padding: 10px;
    border-radius: 5px;
    border-color: white;
}

@media (min-width: 1200px) {
    .quiz {
    max-width: 1140px;
    }
}
@media (min-width: 992px) and (max-width: 1200px) {
    .quiz {
    max-width: 960px;
    }
}

@media (min-width: 768px) and (max-width: 992px) {
    .quiz {
    max-width: 720px;
    }
}
@media (min-width: 576px) and (max-width: 768px)  {
    .quiz {
    max-width: 540px;
    }
}
