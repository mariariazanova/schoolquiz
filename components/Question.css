.quiz-question {
    display: flex;
    flex-direction: column;
    background-color:#98FB98;
}
select[name="question-type"] {
    width: 200px;
}
.check-answers {
    display: flex;
    justify-content: space-between;
}
.check-answers input {
    margin-bottom: 6px;
}
.check-answers input[type="checkbox"] {
    margin-top: 0px;
    width: 20px;
    height: 20px;
}


input[name="answers"] {
    width: 95%;
}