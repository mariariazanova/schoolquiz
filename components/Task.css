.all-tasks .one-task .cell {
    padding: 10px 15px;
}