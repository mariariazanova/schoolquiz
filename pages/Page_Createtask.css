.create-task-page {
    display: flex;
    min-height: 200px;
    background-color: linen;
    border-radius: 5px;
    padding:  10px 10px;

}

.create-task-page-block {
    flex-grow: 5;
    margin-right: 10px;
    min-height: 200px;
    background-color: lightyellow;
    border-radius: 5px;
    padding:  10px 10px;

}
.create-task-page_inner {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

}
.create-task-page .aside-right {
    flex-grow: 1;

    background-color: lightyellow;
    border-radius: 5px;
    padding:  10px 10px;
}