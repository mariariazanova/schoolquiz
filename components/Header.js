import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Link, BrowserRouter, Route  } from 'react-router-dom'
import Login from '../pages/Login';

import './Header.css';
import FormRequest from '../components/FormRequest';
import img from '../images/logo.png';




class Header extends React.Component {

    state = {
       showForm: false
    }

    showRegForm = (EO) => {
       this.setState ({showForm: true})
    } 
     
    render() {
    
      const navBarItemsList = this.props.navbar.map((item, index) => (
        <li key={index} className={"page-item"}>
          <a href="/#" className="page-link">{item}</a>
        </li>

            
      ));
      return (
      <React.Fragment>
        <div className="quiz">  
          <div className="header">
            <div className="logo">
              <img src={img} className="logo_img" />
            </div>
            <BrowserRouter>
            <div className="registration_block">
              <div className="" onClick={this.showRegForm}>               
              Форма обратной связи
              </div>
            </div> 
            </BrowserRouter>
          </div>
        </div>
        {
        (this.state.showForm)
        ? <FormRequest />
        : null
         }
      </React.Fragment>                  
      );  
    } 
  }
  
export default Header;