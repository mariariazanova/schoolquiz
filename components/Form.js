import React, { useFormFields } from 'react';
import ReactDOM from 'react-dom';
import PropTypes, { number } from 'prop-types';
import { auth, database, tasks } from "../firebase";

import './Form.css';
import './styles.css';
import Question from './Question.js';


var subjects = ['русский язык', 'русская литература', 'белорусский язык', 'белорусская литература', 'математика', 'английский язык', 'биология', 'география', 'физика','химия'];
var forms = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];

class Form extends React.Component {

    state = {
       title: '',
       subject: '',
       form: '',
       questions: [],
       question: false,
       countquestion: 1,
       questiontype : '',
       answers: [], 
       correctanswers: [],
       questionsInfo: [],
       quizInfo: {},
       id: 1,
    } 
    
    

    titleChange = (EO) => {
        this.setState({title: EO.target.value});
        console.log(this.state.title);
      }
    subjejctSelect =(EO) => {
        this.setState({subject: EO.target.value});
        console.log(this.state.subject);
    }  
    formSelect = (EO) => {
        this.setState({form: EO.target.value});
        console.log(this.state.form);
    }
    addQuestion = (EO)  => {
        const questions = this.state.questions.concat(
        <Question key={this.state.countquestion} cbQuestionType={this.setQuestionType}
                  cbAnswers={this.setAnswers} cbQuestion={this.setQuestion}
                  countquestion={this.state.countquestion} cbCorrectAnswers={this.setCorrectAnswer}
                  cbApproveQuestion={this.saveQuestion}
        />);
        this.setState({questions: questions,
                      countquestion: this.state.countquestion+1 
        });
       
    }
    /*    
    setQuestionType = (questiontype)  => {
       this.setState({questiontype: questiontype});
       console.log(this.state.questiontype);
    }
    setQuestion = (question)  => {
      this.setState({question: question});
      console.log(this.state.question);
    }
    setAnswers  = (answers)  => {
       this.setState({answers: answers});
       console.log(this.state.answers);
    }
    setCorrectAnswer = (correctanswer)  => {
      this.setState({correctanswers: correctanswer});
      console.log(this.state.correctanswers);
   }
  */

   saveQuestion = (questiontype, question, answers, correctanswers) => {
      this.setState({questiontype: questiontype, question: question,
                    answers: answers, correctanswers: correctanswers
      });
      const questionInfo = {questiontype: questiontype,
        question: question, answers: answers,
        correctanswers: correctanswers}; 

      const questionsInfoNew = this.state.questionsInfo.concat(questionInfo); 
      this.setState({questionsInfo: questionsInfoNew}); 
      //console.log(this.state.questionsInfo);
      //console.log(JSON.stringify(questionsInfoNew));


   } 

    handleSubmit = (EO) => {
   
        const quizInfoNew = {id: this.state.id, title: this.state.title, 
                            subject:this.state.subject, 
                            form: this.state.form, questions: this.state.questionsInfo};
        this.setState({quizInfo: quizInfoNew, id: this.state.id + 1});                   
                      
        console.log(JSON.stringify(quizInfoNew));
        const quizInfoReady=JSON.stringify(quizInfoNew);

        EO.preventDefault(quizInfoReady);

        tasks.push(quizInfoNew);
        //tasks.push(quizInfoReady);


      }

      
    render() {
    

        return (
           <React.Fragment>
            <form id="quiz-form" onSubmit={this.handleSubmit}>
              <h2>Форма для создания теста:</h2>  
              <p>(Заполните все предоставленые поля)</p>
              <div className="quiz-details">
                <div className="labels-list">
                  <label htmlFor="title">Название теста:</label>
                  <label htmlFor="subject">Предмет:</label>
                  <label htmlFor="form">Класс:</label>
                </div>
                <div className="inputs-list">
                  <input type="text" name="title" value={this.state.title} onChange={this.titleChange} placeholder="Например:Правописание безударных гласных"/>
                  <select name="subject" value={this.state.subject} onChange={this.subjejctSelect}>
                    <option key="sub">Выберите предмет</option>
                        {subjects.map(function (item) {
                        return <option key={item} value={item}>{item}</option>;
                        })
                    }
                  </select>
                  <select name="form" value={this.state.form} onChange={this.formSelect}>
                    <option key="form">Выберите класс</option>
                     {forms.map(function (item) {
                        return <option key={item} value={item}>{item}</option>;
                       })
                     }
                  </select>
                </div>
              </div>
              <div>
                    <button type="button" onClick={this.addQuestion}>Создать вопрос №{this.state.countquestion}</button>
                    </div>  
                  <div className="inputs">
                       { this.state.questions }
                  </div>
                  
                  
               
              <button type="submit">Отправить</button>   
            </form>
           </React.Fragment>

        );
    }

}   



export default Form; 