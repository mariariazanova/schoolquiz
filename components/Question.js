import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

import './Question.css';
import './styles.css';

var questiontypes = ['одиночный выбор', 'множественный выбор', 'с изображением', 'напиши сам'];

{/*function SubjectItem (props) {
     return 
       ( <option>{this.props.value}</option>);
  }

*/}  


class Question extends React.Component {

    state = {
        questiontype: ''
    }    

    questiontypeSelect = (EO) => {
        this.setState({questiontype: EO.target.value});
        console.log(this.state.questiontype);
    }    

    handleSubmit = (EO) => {
       this.setState({show: this.state.questiontype})
    }

    render() {
    {/*
    const subjectItems = questiontypes.map((item) =>
      <SubjectItem key={item.toString()} value={item} />
    );
    */}
    return (
            
                <React.Fragment>
                <div className="quiz-question">
                <label htmlFor="question-type">Тип вопроса:</label>
                <select name="subject" value={this.state.questiontype} onChange={this.questiontypeSelect}>
                    <option key="type">Выберите тип вопроса</option>
                   {/* <option key="type1">одиночный выбор</option>
                    <option key="type2">множественный выбор</option>
                    <option key="type3">с изображением</option>
                    <option key="type4">напиши сам</option>
                   */}
                   {questiontypes.map((item) =>
                       <option key={item} value={item}>{item}</option>
                        )
                    }
                
                
                </select>
                <input type="submit" value="Выбрать" onSubmit={this.handleSubmit}/>
                </div>

                {(this.state.show)
                
                }
                </React.Fragment>
        );
    }

}
export default Question; 
