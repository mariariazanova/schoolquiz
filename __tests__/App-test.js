import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../components/Header';
import Task from '../components/Task';
import TaskView from '../components/TaskView';
import Question from '../components/Question';
import Form from '../components/Form';

import PagesLinks from '../pages/PagesLinks';
import Page_Useful from '../pages/Page_Useful';
import Page_Main from '../pages/Page_Main';
import Page_Createtask from '../pages/Page_Createtask';


import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import publicInfo from '../publicInfo.json';
import schoolquiz from '../schoolquiz1.json';
import schoolquiz11 from '../schoolquiz11.json';

test('внешний вид компонента Header', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
          <Header navbar={publicInfo.NavBarItems} 
          main={publicInfo.Main}
          />
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
     
    //имитируем отображение формы обратной связи из Header
    component.getInstance().showRegForm();
    // получаем снэпшот со всеми клиентами
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot()

});

test('внешний вид компонента Task', () => {

      // создаём тестовую версию компонента
      const component = renderer.create(
            <Task 
            title={schoolquiz.title} subject={schoolquiz.subject} 
            form={schoolquiz.form} id={schoolquiz.id}
            questions={schoolquiz.questions} 
            />
      );
 
      // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
      let componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot();

  });

  test('внешний вид компонента TaskView', () => {

      // создаём тестовую версию компонента
      const component = renderer.create(
            <TaskView  info={schoolquiz11} 
            />
      );
 
      // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
      let componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot();

  });

  test('внешний вид компонента Question', () => {

      // создаём тестовую версию компонента
      const component = renderer.create(
            <Question  key={1} 
            />
      );
 
      // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
      let componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot();

  });

  test('внешний вид компонента Form', () => {

      // создаём тестовую версию компонента
      const component = renderer.create(
            <Form   
            />
      );
 
      // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
      let componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot();

      //имитируем отображение сохранения теста 
      component.getInstance().handleSubmit();
      // получаем снэпшот со всеми клиентами

      link.simulate('click', {
            preventDefault: () => {
            }
           });

      componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot()

  });

  test('внешний вид PagesLinks', () => {

      // создаём тестовую версию компонента
      const component = renderer.create(
        <BrowserRouter>
            <PagesLinks />
        </BrowserRouter>
      );
    
      // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
      let componentTree=component.toJSON();
      expect(componentTree).toMatchSnapshot();
  });

  test('внешний вид Page_Useful', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
      <BrowserRouter>
          <Page_Useful />
      </BrowserRouter>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});

test('внешний вид Page_Main', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
      <BrowserRouter>
          <Page_Main />
      </BrowserRouter>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});


test('внешний вид Page_Createtask', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
     <Provider>  
      <BrowserRouter>
          <Page_Createtask />
      </BrowserRouter>
      </Provider>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});
