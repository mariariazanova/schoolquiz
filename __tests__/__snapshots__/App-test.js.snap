// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид Page_Main 1`] = `
Array [
  <div
    className="main_mainpage"
  >
    <div
      className="main_mainpage_asideleft"
    >
      <h3>
        Новости
      </h3>
    </div>
    <div
      className="main_mainpage_center"
    >
      <p>
        Здравствуйте, дорогие друзья! Перед вами приложение для обучения и тестирования знаний школьников. У нас есть готовые тесты для школьников 1-11 классов по многим предметам. Кроме того, вы можете создавать свои тесты. Чтобы получить доступ к полному функционалу, зарегистрируйтесь!
      </p>
      <p>
        Тесты составлены на основании учебных программ по учебным предметам для учреждений общего среднего образования с русским языком обучения и воспитания.
      </p>
      <div
        className="main_mainpage_button"
      >
        <input
          name=""
          onClick={[Function]}
          type="button"
          value="Узнать больше"
        />
      </div>
    </div>
    <div
      className="main_mainpage_asideright"
    >
      <h3>
        Контакты
      </h3>
    </div>
  </div>,
  <div>
    <div
      className="main_add_no"
    >
      <p>
        Мы очень хотели сделать это приложение полезным и интересным как для учителей, так и для школьников. Надеемся, у нас получилось.
      </p>
      <p>
        Школьники могут проходить тесты без регистрации. Планируется реализовать возможность сохранять результаты пройденных тестов после регистрации ребенка.
      </p>
      <p>
        Учителя также могут иметь доступ ко всем тестам на сайте. Кроме того, каждый учитель может создать свой тест по любому предмету! Макет для создания теста достаточно прост и удобен, но в то же время обладает большим функционалом. Вы можете создавать различные виды тестов (например, текстовый вопрос - выбери один из ответов, текстовый вопрос - запиши ответ, угадай по картинке, соотнеси объекты и т.д.)! После одобрения теста администратом ваш тест станет виден (на выбор только тем, кому вы предоставите на него ссылку, или всем пользователям приложения).
      </p>
      <div>
        <img
          src="../images/school.png"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`внешний вид Page_Useful 1`] = `
<div
  className="usefulpage"
>
  <div
    className="usefulpage_center"
  >
    <div
      className="tasks-table"
    >
      <h3 />
    </div>
  </div>
</div>
`;

exports[`внешний вид PagesLinks 1`] = `
<div
  className="navbar"
>
  <a
    aria-current={null}
    className="PageLink"
    href="/"
    onClick={[Function]}
    style={undefined}
  >
    Главная
  </a>
  <a
    aria-current={null}
    className="PageLink"
    href="/tasks"
    onClick={[Function]}
    style={undefined}
  >
    Тесты
  </a>
  <a
    aria-current={null}
    className="PageLink"
    href="/useful"
    onClick={[Function]}
    style={undefined}
  >
    Полезная информация
  </a>
  <a
    aria-current={null}
    className="PageLink"
    href="/checklogin"
    onClick={[Function]}
    style={undefined}
  >
    Создать свой тест
  </a>
  <a
    aria-current={null}
    className="PageLink"
    href="/login"
    onClick={[Function]}
    style={undefined}
  >
    Зарегистрироваться
  </a>
</div>
`;

exports[`внешний вид компонента Form 1`] = `
<form
  id="quiz-form"
  onSubmit={[Function]}
>
  <h2>
    Форма для создания теста:
  </h2>
  <p>
    (Заполните все предоставленые поля)
  </p>
  <div
    className="quiz-details"
  >
    <div
      className="labels-list"
    >
      <label
        htmlFor="title"
      >
        Название теста:
      </label>
      <label
        htmlFor="subject"
      >
        Предмет:
      </label>
      <label
        htmlFor="form"
      >
        Класс:
      </label>
    </div>
    <div
      className="inputs-list"
    >
      <input
        name="title"
        onChange={[Function]}
        placeholder="Например:Правописание безударных гласных"
        type="text"
        value=""
      />
      <select
        name="subject"
        onChange={[Function]}
        value=""
      >
        <option>
          Выберите предмет
        </option>
        <option
          value="русский язык"
        >
          русский язык
        </option>
        <option
          value="русская литература"
        >
          русская литература
        </option>
        <option
          value="белорусский язык"
        >
          белорусский язык
        </option>
        <option
          value="белорусская литература"
        >
          белорусская литература
        </option>
        <option
          value="математика"
        >
          математика
        </option>
        <option
          value="английский язык"
        >
          английский язык
        </option>
        <option
          value="биология"
        >
          биология
        </option>
        <option
          value="география"
        >
          география
        </option>
        <option
          value="физика"
        >
          физика
        </option>
        <option
          value="химия"
        >
          химия
        </option>
      </select>
      <select
        name="form"
        onChange={[Function]}
        value=""
      >
        <option>
          Выберите класс
        </option>
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
        <option
          value={6}
        >
          6
        </option>
        <option
          value={7}
        >
          7
        </option>
        <option
          value={8}
        >
          8
        </option>
        <option
          value={9}
        >
          9
        </option>
        <option
          value={10}
        >
          10
        </option>
        <option
          value={11}
        >
          11
        </option>
      </select>
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Создать вопрос №
      1
    </button>
  </div>
  <div
    className="inputs"
  />
  <button
    type="submit"
  >
    Отправить
  </button>
</form>
`;

exports[`внешний вид компонента Header 1`] = `
<div
  className="quiz"
>
  <div
    className="header"
  >
    <div
      className="logo"
    >
      <img
        className="logo_img"
        src="test-file-stub"
      />
    </div>
    <div
      className="registration_block"
    >
      <div
        className=""
        onClick={[Function]}
      >
        Форма обратной связи
      </div>
    </div>
  </div>
</div>
`;

exports[`внешний вид компонента Header 2`] = `
Array [
  <div
    className="quiz"
  >
    <div
      className="header"
    >
      <div
        className="logo"
      >
        <img
          className="logo_img"
          src="test-file-stub"
        />
      </div>
      <div
        className="registration_block"
      >
        <div
          className=""
          onClick={[Function]}
        >
          Форма обратной связи
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="new_form"
    >
      <h1>
        Форма обратной связи:
      </h1>
      <form
        onSubmit={undefined}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Введите свой email:
          </label>
          <input
            aria-describedby="emailHelp"
            className="form-control"
            id="email"
            onChange={undefined}
            placeholder="aaaa@mail.ru"
            type="email"
            value={undefined}
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputPassword1"
          >
            Введите свое имя:
          </label>
          <input
            className="form-control"
            id="password"
            onChange={undefined}
            placeholder="abc123"
            type="password"
            value={undefined}
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputRequire"
          >
            Что вы хотите нам сказать:
          </label>
          <textarea
            className="form-control"
            cols="45"
            id="password"
            name="text"
            onChange={undefined}
            placeholder="Мне очень понравилось ваше приложение!"
            rows="10"
            value={undefined}
          />
        </div>
        <button
          className="btn-submit"
          type="submit"
        >
          Отправить
        </button>
      </form>
    </div>
  </div>,
]
`;

exports[`внешний вид компонента Question 1`] = `
Array [
  <div
    className="quiz-question"
  >
    <div>
      <h4>
        Вопрос №
      </h4>
      <label
        htmlFor="question-type"
      >
        Тип вопроса:
      </label>
      <select
        name="question-type"
        onChange={[Function]}
        value=""
      >
        <option>
          Выберите тип вопроса
        </option>
        <option
          value="одиночный выбор"
        >
          одиночный выбор
        </option>
        <option
          value="множественный выбор"
        >
          множественный выбор
        </option>
      </select>
    </div>
    <label
      htmlFor="question"
    >
      Введите вопрос:
    </label>
    <input
      name="question"
      onChange={[Function]}
      placeholder="Например: 3 + 2 ="
      type="text"
      value=""
    />
    <label
      htmlFor="answers"
    >
      Введите варианты ответов и отметьте галочкой правильные ответы:
    </label>
    <div
      className="check-answers"
    >
      <input
        name="answers"
        onChange={[Function]}
        placeholder="Например: 5"
        type="text"
        value=""
      />
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="check-answers"
    >
      <input
        name="answers"
        onChange={[Function]}
        placeholder="Например: 4"
        type="text"
        value=""
      />
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="check-answers"
    >
      <input
        name="answers"
        onChange={[Function]}
        placeholder="Например: 3"
        type="text"
        value=""
      />
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="check-answers"
    >
      <input
        name="answers"
        onChange={[Function]}
        placeholder="Например: не знаю"
        type="text"
        value=""
      />
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>,
  <button
    onClick={[Function]}
    type="button"
  >
    Сохранить вопрос
  </button>,
]
`;
